safe-global/safe-wallet-web
# fix: Show approval editor for ERC721 approvals

## What it solves

Resolves #3774 

## How this PR fixes it

- When the ERC20 token info request fails we try the ERC721 contract and fetch the symbol, if that fails we check the ERC-165 function `supportsInterface` with the ERC721 identifier to at least know if its an ERC721 token or not.

Caveat: If its a multisend that contains both ERC20 and ERC721 approvals all of them will be read-only now.

## How to test it

1. Open the transaction build with an ERC721 contract address
2. Create an `approve` transaction and continue
3. Observe no error messages in the approval editor

## Screenshots
<img width="705" alt="Screenshot 2024-07-26 at 11 41 04" src="https://github.com/user-attachments/assets/c6ff9c53-c009-49a6-8fcc-4d9b100fcd2d">


## Checklist
